src/main/java/weizberg/earthquake/EarthquakeFrame.java

```diff
@@ -0,0 +1,148 @@
+                .subscribe(
+                        (response) -> handleResponse(response, jlistOneMonth),
+                        Throwable::printStackTrace);
+
+        jlistOneHour.addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                if (!listSelectionEvent.getValueIsAdjusting()) {
+                    int indexOfSelected = jlistOneHour.getSelectedIndex();
+                    if (indexOfSelected >= 0) {
+                        try {
+                            Desktop.getDesktop().browse(new URI("https://maps.google.com/?q=" + longAndLat[indexOfSelected]));
+                        }
+                        catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            }
+        });
+
+        jlistOneMonth.addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                if (!listSelectionEvent.getValueIsAdjusting()) {
+                    int indexOfSelected = jlistOneMonth.getSelectedIndex();
+                    if (indexOfSelected >= 0) {
+                        try {
+                            Desktop.getDesktop().browse(new URI("https://maps.google.com/?q=" + longAndLat[indexOfSelected]));
+                        }
+                        catch (Exception e){
+                            e.printStackTrace();
+                        }
+                    }
+                }
+            }
+        });
+
+        oneHourButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent actionEvent) {
+                addOneHour();
+            }
+        });
+
+        oneMonthButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent actionEvent) {
+                addOneMonth();
+            }
+        });
+
+
+    }
+
+    private void addOneHour() {
+        getContentPane().add(jlistOneHour, BorderLayout.CENTER);
+        revalidate();
+        repaint();
+    }
+
+    private void addOneMonth() {
+        getContentPane().add(jlistOneMonth, BorderLayout.CENTER);
+        revalidate();
+        repaint();
+    }
+
+
+    private void handleResponse(FeatureCollection response, JList<String> jlist) {
+
+        String[] listData = new String[response.features.length];
+        longAndLat = new String[response.features.length];
+        for (int i = 0; i < response.features.length; i++) {
+            Feature feature = response.features[i];
+            listData[i] = feature.properties.mag + " " + feature.properties.place;
+            longAndLat[i] = feature.geometry.coordinates[0] + "," + feature.geometry.coordinates[1];
```

Comment:
Save the FeatureCollection to be used later instead of creating all these Strings now.